docs/apps/components/data-grid.md

```diff
@@ -0,0 +1,208 @@
+# Data Grid
+
+## Description
+
+**Data Grid** is a tabular data display component. It enables you to add different functionalities.
+
+<img src= "/apps/components/img/datagrid.png">
+
+## Use Case
+
+1. Displaying and Editing Tabular Data.
+2. Collecting and Managing Lists.
+3. Building Dynamic Forms.
+4. Integrating with External Data.
+
+### Interactivity
+
+1. Allow users to directly edit data within cells, eliminating the need for separate edit forms or modals.
+2. Provide tools for users to sort data by different columns and apply filters to narrow down the displayed information.
+3. Pagination
+4. Expandable Rows
+5. Enforce data integrity by implementing validation rules to ensure data accuracy and consistency.
+
+## API
+
+### Events
+
+| **Name**| **Description**|
+|---------|----------------|
+| `onCellClicked`| The onCellClicked event occurs when the cell in a data grid is clicked|
+| `onLinkClicked`| The onLinkClicked event occurs when the link is clicked|
+
+### Methods
+
+| **Name**| **Description**|**Parameters**|
+|---------|----------------|--------------|
+|`this.addClassName`|Add style class to a form item|`(fields: String \| String[], className: String)`|
+|`this.removeClassName`|Remove style class from a form item|`(fields: String \| String[], className: String)`|
+|`this.getComponent`|Returns a component whose id has been passed as a parameter|`(component_ID: String : Object)`|
+|`this.getValues`|Get the values of all fields when values change|`() : Object`|
+|`this.hide`|Hides the field|`(fields: String \| String[])`|
+|`this.show`|Displays the field|`(fields: String \| String[])`|
+|`this.disable`| Disable input field from user interaction|`(fields: String \| String[])`|
+|`this.getValue`|Get A Value From a component|`(fieldName: String)`|
+|`this.setData`|Set the data in the field|`(Value: Object)`|
+
+!!! Info
+    1. The show() and hide() methods can also be used to control the visibility of an alert in response to user input.
+    2. **Fields** refers to a **component ID** or a **list of component IDs**. You can fetch the ID from the **Component Attribute** panel in the Page Builder.
+    3. Before using **this.show(fields)**, make sure the component is **hidden**. This can be done using **this.hide(fields)** or by **enabling** the **Hidden checkbox** in the Component Attribute panel.
+
+#### Steps to use the methods for the Page Builder components
+
+1. Go to Form Attribute :material-menu-right: Action Panel :material-menu-right: Setting :material-menu-right: (Mounted | refresh | click 'Add action').
+2. Write the method/code as shown in the **Example** below.
+3. Click on `Save`.
+4. On the main screen click on `Save` again.
+5. Click on `Preview` to see the code in action.
+<img src= "/apps/components/img/check1.png">
```

Comment:
An image showing reference to a data grid would be more appropriate here instead of reference to a checkbox. Also considering that the disable function is not effective on this component. It could be better to use a function that is effective such as getComponent or getValue

Reply:
done